### PR TITLE
feat: implemented trigger conditions feature

### PR DESCRIPTION
**Pull request by Toxer10**
image

### CODE REVIEW

**Review by haneabogdan — APPROVED** [review]

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import {SymbolWrapper} from '@custom-antd/Form/Form.styled';
+import FormItemLabel from '@custom-antd/Form/FormItem/FormItemLabel';
+
+import {ConfigurationCard, notificationCall} from '@molecules';
+
+import {Permissions, usePermission} from '@permissions/base';
+
+import {useUpdateTriggerByIdMutation} from '@services/triggers';
+
+import {TestTrigger} from '@src/models/triggers';
```
[the remark this answers is not in the document]
Comment (Toxer10) [reply src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:14]: Agree ✅

File: src/components/custom-antd/Form/Form.styled.tsx [hunk 2]
```diff
@@ -11,8 +11,13 @@ export const StyledFormItemLabel = styled.label`
   font-weight: 300;
 `;
 
-export const StyledFormItem = styled(AntdFormItem)`
-  flex: 1;
+export const StyledFormItem = styled(AntdFormItem)<{flex?: number}>`
+  flex: ${({flex}) => flex};
```
[the remark this answers is not in the document]
Comment (rangoo94) [reply src/components/custom-antd/Form/Form.styled.tsx:15]: Probably worth to either rename it to `flexGrow` or allow string too

File: src/components/custom-antd/Form/FormRow/FormRow.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import {PropsWithChildren} from 'react';
```
Comment (rangoo94) [inline src/components/custom-antd/Form/FormRow/FormRow.tsx:1]: Looks like it's not custom Antd component, as it's not using anything from Antd

Comment (Toxer10) [reply src/components/custom-antd/Form/FormRow/FormRow.tsx:1]: It's used in `<Form />`. I would store this component in `@custom-antd`.

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                <FormItemLabel
+                  text="Delay in ms"
+                  tooltipMessage="Add the delay in ms the test trigger waits for conditions"
+                />
+              }
+            >
+              <InputNumber
+                controls={false}
+                placeholder="Delay"
+                max={2147483647} // Int32 max value
```
[the remark this answers is not in the document]
Comment (Toxer10) [reply src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:95]: Can be

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                                      value: 'ProgressDeadlineExceeded',
+                                    },
+                                  ]}
+                                  placeholder="Type"
+                                />
+                              </FormItem>
+                              <FormItem {...restField} name={[name, 'status']} rules={[requiredNoText]} flex={2}>
+                                <Select
+                                  options={[
+                                    {label: 'True', value: 'True'},
```
[the remark this answers is not in the document]
Comment (Toxer10) [reply src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:137]: How can we use `TriggerConditionStatus` enum as value for `<Select />`?
Like this? @rangoo94 
image

[the remark this answers is not in the document]
Comment (rangoo94) [reply src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx]: Yeah, exactly, either:
* this way
* or keeping the `label` as string (i.e. `{label "True", value: TriggerConditionStatus.True}`)
* or building it (i.e. `[TriggerConditionStatus.True, TriggerConditionStatus.False, TriggerConditionStatus.Unknown].map(value => ({label: value, value}))`)

File: src/components/custom-antd/Form/Form.styled.tsx [hunk 6]
```diff
@@ -11,8 +11,13 @@ export const StyledFormItemLabel = styled.label`
   font-weight: 300;
 `;
 
-export const StyledFormItem = styled(AntdFormItem)`
-  flex: 1;
+export const StyledFormItem = styled(AntdFormItem)<{$flexGrow?: number | string}>`
```
Comment (rangoo94) [inline src/components/custom-antd/Form/Form.styled.tsx:14]:
```suggestion
export const StyledFormItem = styled(AntdFormItem)<{$flexGrow: number | string}>`
```

File: src/components/custom-antd/Form/FormItem/FormItemLabel.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import {QuestionCircleOutlined} from '@ant-design/icons';
```
Comment (rangoo94) [inline src/components/custom-antd/Form/FormItem/FormItemLabel.tsx:1]: We have it already in `src/components/molecules/GitFormItems/FormItemLabel.tsx` - could you delete the one in `GitFormitems` and use the one here instead?

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 8]
```diff
@@ -0,0 +1,162 @@
+import {DeleteOutlined} from '@ant-design/icons';
+import {Form, Input, InputNumber, Select} from 'antd';
+
+import {Button, FormItem, FormRow, FullWidthSpace} from '@custom-antd';
+import {SymbolWrapper} from '@custom-antd/Form/Form.styled';
+import FormItemLabel from '@custom-antd/Form/FormItem/FormItemLabel';
```
Comment (rangoo94) [inline src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:6]: Probably it should be exposed in the `@custom-antd/Form/index.ts` barrel

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 9]
```diff
@@ -0,0 +1,162 @@
+import {DeleteOutlined} from '@ant-design/icons';
+import {Form, Input, InputNumber, Select} from 'antd';
+
+import {Button, FormItem, FormRow, FullWidthSpace} from '@custom-antd';
+import {SymbolWrapper} from '@custom-antd/Form/Form.styled';
```
Comment (rangoo94) [inline src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:5]: I'm not sure what it actually does, the name doesn't say anything meaningful (given that some `Delete` button is used here as a child);

* Is the name correct?
* Should it be part of `@custom-antd/Form`, or maybe it's not connected?
* Should it be generic, or rather local for `ResourceCondition`?

File: src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+                              options={triggersKeyMap.conditions.map(condition => ({
+                                label: condition,
+                                value: condition,
+                              }))}
+                              placeholder="Type"
+                            />
+                          </FormItem>
+                          <FormItem {...restField} name={[name, 'status']} rules={[requiredNoText]} flex={2}>
+                            <Select
+                              options={[
```
Comment (rangoo94) [inline src/components/pages/Triggers/TriggerDetails/TriggerSettings/TriggerCondition/ResourceCondition.tsx:120]: The options could be put on top to separate variable and memoized

File: src/components/custom-antd/Form/FormItem/FormItem.tsx [hunk 11]
```diff
@@ -2,12 +2,17 @@ import {FormItemProps} from 'antd';
...
-  const {children, label, ...rest} = props;
+type FormItemCustomProps = {
+  flex?: number | string;
+};
+
+const FormItem: React.FC<FormItemProps & FormItemCustomProps> = props => {
+  const {children, label, flex = 1, ...rest} = props;
 
-  const formItemsProps: FormItemProps = {
+  const formItemsProps: FormItemProps & {$flexGrow?: number | string} = {
```
[the remark this answers is not in the document]
Comment (Toxer10) [reply src/components/custom-antd/Form/FormItem/FormItem.tsx]: It's already required
image

**Pull request merged**